internal/debug/flags.go

```diff
@@ -141,30 +146,37 @@ func init() {
 // Setup initializes profiling and logging based on the CLI flags.
 // It should be called as early as possible in the program.
 func Setup(ctx *cli.Context) error {
-	logFile := ctx.String(logFileFlag.Name)
-	useColor := logFile == "" && os.Getenv("TERM") != "dumb" && (isatty.IsTerminal(os.Stderr.Fd()) || isatty.IsCygwinTerminal(os.Stderr.Fd()))
+	var ostream log.Handler
+	output := io.Writer(os.Stderr)
 
-	var logfmt log.Format
+	var fmtr log.Format
 	if ctx.Bool(logjsonFlag.Name) {
-		logfmt = log.JSONFormat()
+		fmtr = log.JSONFormat()
 	} else {
-		logfmt = log.TerminalFormat(useColor)
+		usecolor := (isatty.IsTerminal(os.Stderr.Fd()) || isatty.IsCygwinTerminal(os.Stderr.Fd())) && os.Getenv("TERM") != "dumb"
+		if usecolor {
+			output = colorable.NewColorableStderr()
+		}
+		fmtr = log.TerminalFormat(usecolor)
 	}
 
-	if logFile != "" {
-		var err error
-		logOutputStream, err = log.FileHandler(logFile, logfmt)
-		if err != nil {
-			return err
-		}
+	stdHandler := log.StreamHandler(output, fmtr)
+	locationURL, err := getLogLocationURL(ctx)
+	if err != nil {
+		log.Warn("issue with initiatilizing file logger", "err", err)
+		ostream = stdHandler
```

Comment:
Is this message displayed anywhere? I mean, we're in the middle of setting up the logger. Might be it's already default-setup, in which case this is fine

Reply:
that's right. It's default setup which writes to stdout but not to the file.